src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageItemViewModel.cs

```diff
@@ -678,6 +703,46 @@ private async System.Threading.Tasks.Task ReloadPackageDeprecationAsync()
             }
         }
 
+        //private async Task ReloadPackageVulnerabilitiesAsync()
+        //{
+        //    CancellationToken cancellationToken = _cancellationTokenSource.Token;
+        //    try
+        //    {
+        //        var identity = new PackageIdentity(Id, Version);
+        //        IReadOnlyCollection<PackageVulnerabilityMetadataContextInfo> vulnerabilities = await _searchService.GetVulnerabilityMetadataAsync(identity, Sources, IncludePrerelease, cancellationToken);
+
+        //        await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+        //        cancellationToken.ThrowIfCancellationRequested();
+
+        //        VulnerabilityMaxSeverity = vulnerabilities?.Max(v => v.Severity) ?? -1;
+        //    }
+        //    catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
+        //    {
+        //        // UI requested cancellation
+        //    }
+        //}
+
+        private async Task ReloadPackageMetadataAsync()
+        {
+            CancellationToken cancellationToken = _cancellationTokenSource.Token;
+            try
+            {
+                var identity = new PackageIdentity(Id, Version);
+                (PackageSearchMetadataContextInfo packageMetadata, PackageDeprecationMetadataContextInfo deprecationMetadata) =
+                    await _searchService.GetPackageMetadataAsync(identity, Sources, IncludePrerelease, cancellationToken);
```

Comment:
Considering that we [already do this for deprecation data](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs#L193), is there any reason not to use a single query to retrieve both deprecation & vulnerabilities? My concern is doubling space/time complexity and there's no reason clear to me why we need to.